### PR TITLE
Bugfix api templates

### PR DESCRIPTION
**Pull request by gpaulos**
The api/x.x/templates command has been revised since the original tests were checked in. 

This PR applies corrections to these tests.

The GET api/x.x/templates command is no longer valid without a name spec.

@hohene @johren @jimturnquist @diontje @larry-dean

### CODE REVIEW

File: test/tests/rackhd11/test_rackhd11_api_templates.py [hunk 1]
```diff
@@ -26,17 +29,13 @@ def test_api_11_templates_library(self):
...
-                if fit_common.VERBOSITY >= 2:
-                    print "Checking:", item['name'], subitem
-                self.assertGreater(len(item[subitem]), 0, subitem + ' field error')
+    def test_api_11_templates_library_ID_put_get_delete(self):
+            api_data = fit_common.rackhdapi("/api/1.1/templates/library/testid?scope=global", action="text-put", payload="null")
+            self.assertEqual(api_data['status'], 200, "Was expecting code 200. Got " + str(api_data['status']))
+            api_data = fit_common.rackhdapi("/api/1.1/templates/library/testid")
+            self.assertIn(api_data['json']['contents'], "null", "Data 'null' was not returned.")
+            self.assertEqual(api_data['status'], 200, "Was expecting code 200. Got " + str(api_data['status']))
+            api_data = fit_common.rackhdapi("/api/1.1/templates/library/testid", action="delete")
```
Comment (johren) [inline test/tests/rackhd11/test_rackhd11_api_templates.py]: Should have a check after the delete to make sure that it returned the expected status code and that the item actually went away.

File: test/tests/rackhd11/test_rackhd11_api_templates.py [hunk 2]
```diff
@@ -16,6 +16,9 @@
 from nose.plugins.attrib import attr
 @attr(all=True, regression=True, smoke=True)
 class rackhd11_api_templates(fit_common.unittest.TestCase):
+    def setUp(self):
+        fit_common.rackhdapi("/api/1.1/templates/library/testid", action="delete")
```
Comment (johren) [inline test/tests/rackhd11/test_rackhd11_api_templates.py]: I'm assuming this is to make sure that the id you are going to use in the tests doesn't already exist.  If so, can you add a comment so it is clear to others editing this test later?

File: test/tests/rackhd20/test_rackhd20_api_templates.py [hunk 3]
```diff
@@ -17,7 +17,8 @@
 @attr(all=True, regression=True, smoke=True)
 class rackhd20_api_templates(fit_common.unittest.TestCase):
     def setUp(self):
-        api_data = fit_common.rackhdapi("/api/2.0/templates/library/testid", action="delete")
+        fit_common.rackhdapi("/api/2.0/templates/library/testid", action="delete")
```
Comment (johren) [inline test/tests/rackhd20/test_rackhd20_api_templates.py]: Same here.  Please add a comment to indicate what this is doing.

File: test/tests/rackhd11/test_rackhd11_api_templates.py [hunk 4]
```diff
@@ -26,17 +29,13 @@ def test_api_11_templates_library(self):
...
-            lib_data = fit_common.rackhdapi("/api/1.1/templates/library/" + item['name'])
-            self.assertEqual(lib_data['status'], 200, "Was expecting code 200. Got " + str(lib_data['status']))
-            # check required fields
-            for subitem in ['contents', 'createdAt', 'id', 'name', 'updatedAt']:
-                if fit_common.VERBOSITY >= 2:
-                    print "Checking:", item['name'], subitem
-                self.assertGreater(len(item[subitem]), 0, subitem + ' field error')
+    def test_api_11_templates_library_ID_put_get_delete(self):
+            api_data = fit_common.rackhdapi("/api/1.1/templates/library/testid?scope=global", action="text-put", payload="null")
+            self.assertEqual(api_data['status'], 200, "Was expecting code 200. Got " + str(api_data['status']))
```
Comment (johren) [inline test/tests/rackhd11/test_rackhd11_api_templates.py]: Double-checking.  In the 2.0 api tests it checks for status code of 201.  Is the behavior different in 1.1 vs. 2.0?

Comment (gpaulos) [inline test/tests/rackhd11/test_rackhd11_api_templates.py]: Yes, behavior is different in 1.1 and 2.0.

File: test/tests/rackhd11/test_rackhd11_api_templates.py (hunk 2 above)
Comment (gpaulos) [inline test/tests/rackhd11/test_rackhd11_api_templates.py]: OK I will comment.

File: test/tests/rackhd20/test_rackhd20_api_templates.py (hunk 3 above)
Comment (gpaulos) [inline test/tests/rackhd20/test_rackhd20_api_templates.py]: OK

**Comment by panpan0000** [comment]
abort the Jenkins PR Gate.
due to bintray change, all test will fail because can't download static image.
fixing soon.

**Comment by johren** [comment]
test this please

**Pull request merged**